package/build-binary.sh

```diff
@@ -1,20 +1,17 @@
 #!/bin/sh
 
-# This script invokes Stack to do a release build. It is intended to be run on
+# This script invokes Cabal to do a release build. It is intended to be run on
 # on the same system that packages the binaries, right before build-package.sh.
 
 # Fail early if any of the commands below fail.
 set -e
 
-# Download the right version of GHC.
-stack setup
-
-# When running in a virtual machine, if we inherited a dirty Stack working
-# directory from the host machine, be sure to clean it.
-stack clean
+# When running in a virtual machine, if we inherited a dirty working directory
+# from the host machine, be sure to clean it.
+cabal clean
 
 # Build with --split-objs to produce a smaller binary, and enable optimizations.
```

Comment:
This comment is out of date, I think it should either be the good old GHC flag (`-fsplit-objs`) or the Cabal flag below.

Reply:
I don't know if anyone actually uses this script, but apparently `--split-objs` has been removed five years ago and it was replaced with `-split-sections`. So this hasn't worked for a while now. I did just notice that the `cabal list-bin` call doesn't work correctly here since it will return a different path depending on whether optimizations are enabled or not (to avoid rebuilding when switching between debug and release builds). So I fixed that in 79a7d832f78c7fe2f0c4fa79e15e54ab51703259.